### ISSUE TITLE
CardDAV uknown error

### ISSUE THREAD

**Issue by stratacast** [pierre-alain-b/rainloop-nextcloud#2]
I have established CardDAV with rainloop to sync contacts, my CardDAV server being supplied by Nextcloud 11. When I try to sync my contacts I get an "unknown error". I didn't get anymore information beyond that

**Comment by omigeot** [comment]
What kind of CardDAV URL did you specify to Rainloop? What does it look like?

And do you have access to web server's logs?

**Comment by jybeeL** [comment]
Hi,
I have maybe a similar problem.

When I want to synchronize the nextcloud contacts with the Rainloop contacts, I get this error message: "Contact synchronization error".

In the rainloop administration panel, the box "_Allow contact synchronization (with the external CardDAV server)_" is checked.
In the rainloop user panel, the "_Enable remote synchronization_" check box is selected.
The URL address is: "_https://nextcloud.toto.fr/remote.php/dav/addressbooks/users/admin/contacts/_"
I specify, I use this URL with Thunderbird for example and it works.

Where can I see the log for rainloop?

Thank you for your help :)

**Comment by pierre-alain-b** [comment]
Hello jybeeL,
What about this URL: "https://nextcloud.toto.fr/remote.php/carddav/addressbooks/users/admin/contacts/" ?
(carddav instead of dav)
Best
PAB

**Comment by jybeeL** [comment]
Hi pierre-alain-b,
its doesnt work with "https://nextcloud.toto.fr/remote.php/**carddav**/addressbooks/users/admin/contacts/" 

:(

**Comment by pierre-alain-b** [comment]
What do you see in the logs of Nextcloud (/path/to/nextcloud/data/nextcloud.log)? Anything logged while you try to synchronize the contacts from Rainloop?

**Comment by pierre-alain-b** [comment]
You can also activate Rainloop logging here:

`/path/to/nextcloud/data/rainloop-storage/_data_/_default_/configs/application.ini`

```
[logs]
enable = On
```

And then logs are available in 
`/path/to/nextcloud/data/rainloop-storage/_data_/_default_/logs/`

**Comment by jybeeL** [comment]
thx for log :)

The problem is with Curl.

> [10:52:07.037][7697158a] DAV[DATA]: PROPFIND
[10:52:07.038][7697158a] INFO[NOTICE]: exception 'SabreForRainLoop\DAV\Exception' with message '[CURL] Error while making request: Failed to connect to nextcloud.toto.fr port 443: Connection refused (error code: 7)' in /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/SabreForRainLoop/DAV/Client.php:421

Stack trace:

> #0 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/SabreForRainLoop/DAV/Client.php(179): SabreForRainLoop\DAV\Client->request('PROPFIND', '', '<?xml version="...', Array)
#1 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Providers/AddressBook/PdoAddressBook.php(468): SabreForRainLoop\DAV\Client->propFind('', Array, 1)
#2 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Providers/AddressBook/PdoAddressBook.php(643): RainLoop\Providers\AddressBook\PdoAddressBook->checkContactsPath(Object(SabreForRainLoop\DAV\Client), '')
#3 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Providers/AddressBook/PdoAddressBook.php(677): RainLoop\Providers\AddressBook\PdoAddressBook->getDavClient('https://nextclo...', 'admin', '*******', '')
#4 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Providers/AddressBook.php(75): RainLoop\Providers\AddressBook\PdoAddressBook->Sync('toto@toto....', 'https://nextclo...', 'admin', '*******')
#5 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Actions.php(6693): RainLoop\Providers\AddressBook->Sync('toto@toto....', 'https://nextclo...', 'admin', '*******')
#6 [internal function]: RainLoop\Actions->DoContactsSync()
#7 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/ServiceActions.php(172): call_user_func(Array)
#8 [internal function]: RainLoop\ServiceActions->ServiceAjax('')
#9 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Service.php(146): call_user_func(Array, '')
#10 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Service.php(56): RainLoop\Service->localHandle()
#11 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Service.php(79): RainLoop\Service->__construct()
#12 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/handle.php(94): RainLoop\Service::Handle()
#13 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/include.php(225): include('/var/www/nextcl...')
#14 /var/www/nextcloud/apps/rainloop/app/index.php(13): include('/var/www/nextcl...')
#15 /var/www/nextcloud/apps/rainloop/app.php(35): include('/var/www/nextcl...')
#16 /var/www/nextcloud/lib/private/Route/Route.php(155) : runtime-created function(1): require_once('/var/www/nextcl...')
#17 [internal function]: __lambda_func(Array)
#18 /var/www/nextcloud/lib/private/Route/Router.php(299): call_user_func('\x00lambda_2383', Array)
#19 /var/www/nextcloud/lib/base.php(1010): OC\Route\Router->match('/apps/rainloop/...')
#20 /var/www/nextcloud/index.php(40): OC::handleRequest()
#21 {main}
[10:52:07.038][7697158a] INFO[NOTICE]: exception 'RainLoop\Exceptions\ClientException' with message 'ContactsSyncError[140]' in /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Actions.php:6699
Stack trace:
#0 [internal function]: RainLoop\Actions->DoContactsSync()
#1 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/ServiceActions.php(172): call_user_func(Array)
#2 [internal function]: RainLoop\ServiceActions->ServiceAjax('')
#3 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Service.php(146): call_user_func(Array, '')
#4 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Service.php(56): RainLoop\Service->localHandle()
#5 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/libraries/RainLoop/Service.php(79): RainLoop\Service->__construct()
#6 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/app/handle.php(94): RainLoop\Service::Handle()
#7 /var/www/nextcloud/apps/rainloop/app/rainloop/v/1.10.4.183/include.php(225): include('/var/www/nextcl...')
#8 /var/www/nextcloud/apps/rainloop/app/index.php(13): include('/var/www/nextcl...')
#9 /var/www/nextcloud/apps/rainloop/app.php(35): include('/var/www/nextcl...')
#10 /var/www/nextcloud/lib/private/Route/Route.php(155) : runtime-created function(1): require_once('/var/www/nextcl...')
#11 [internal function]: __lambda_func(Array)
#12 /var/www/nextcloud/lib/private/Route/Router.php(299): call_user_func('\x00lambda_2383', Array)
#13 /var/www/nextcloud/lib/base.php(1010): OC\Route\Router->match('/apps/rainloop/...')
#14 /var/www/nextcloud/index.php(40): OC::handleRequest()
#15 {main}
[10:52:07.038][7697158a] AJAX[DATA]: {"Action":"ContactsSync","Result":false,"ErrorCode":140,"ErrorMessage":"","ErrorMessageAdditional":"","Time":24}
[10:52:07.044][7697158a] INFO[MEMORY]: Memory peak usage: 13.75MB
[10:52:07.044][7697158a] INFO[TIME]: Time delta: 0.029705047607422

I search how i can accept my autosigned certificat with curl.

**Comment by pierre-alain-b** [comment]
Thank you for letting us know. You can certainly set curl.cainfo to a sensible value (like pointing to a CA PEM including the .crt generated when self signing) in php.ini.
http://php.net/manual/fr/curl.configuration.php
I will close the issue (as it is a local config issue) but do not hesitate to post the final solution for those passing by.

**Comment by jybeeL** [comment]
thank for your help :)

i modify curl.cainfo in /etc/php/apach2/php.ini
> curl.cainfo = "/etc/ssl/nextcloud.toto.fr/cert.pem"

and /etc/hosts 
> 10.8.8.120      nextcloud nextcloud.toto.fr

and it's okay ^_^
I see the same contacts in rainloop and in the Nextcloud contact application.

**Comment by brdobrdo** [comment]
So if you install Nextcloud on a shared hosting and Rainloop and it doesn't work (contacts don't sync and they do not autocomplete when typing the address), is it still a local configuration issue. If yes, how to configure? If not, why won't it work?

**Comment by pierre-alain-b** [comment]
Please post more information on the issue: what you did, the error messages you see...

**Comment by brdobrdo** [comment]
I have installed Nextcloud using Softaculous on a shared hosting server. Inside Nextcloud I have installed Rainloop. When in Nextcloud contacts app I have copied the link for sharing contacts. Inside Rainloop --> Settings --> Contacts I have set up CardDAV sync by pasting the link I copied and provided username and password (the same I use to login to nextcloud). 
When I try to sync contacts (hamburger icon inside contacts in Rainloop) it says Contacts synchronization error. Also, when typing into address field nothing autocompletes from Nextcloud contacts (like some users reported it does).
Shared hosting is Linux, Nextcloud is 18.0.4.
